src/actions/__tests__/search.test.js

```diff
@@ -28,152 +28,182 @@ const stateWithoutScopeQuery = {
   }),
 };
 
-describe('search - async action creator', () => {
-  afterEach(() => {
-    mockHttp.reset();
+describe('search - action creators', () => {
+  describe('searchForCurrentLocation', () => {
+    it('create SEARCH_REQUEST and SEARCH_SUCCESS if search request is successful', async done => {
+      const store = getStoreWithState({
+        router: {
+          location: { pathname: '/test', search: '?size=10&q=test' },
+        },
+      });
+      mockHttp.onGet('/test?size=10&q=test').replyOnce(200, { foo: 'bar' });
+
+      await store.dispatch(searchForCurrentLocation());
+
+      const expectedActions = [
+        { type: types.SEARCH_REQUEST },
+        { type: types.SEARCH_SUCCESS, payload: { foo: 'bar' } },
+      ];
+      expect(store.getActions()).toEqual(expectedActions);
+      done();
+    });
+
+    it('create SEARCH_REQUEST and SEARCH_ERROR if search request is unsuccessful', async done => {
+      const store = getStoreWithState({
+        router: {
+          location: { pathname: '/test', search: '?size=10&q=test' },
+        },
+      });
+      mockHttp.onGet('/test?size=10&q=test').networkError();
+
+      await store.dispatch(searchForCurrentLocation());
+
+      const expectedActions = [
+        { type: types.SEARCH_REQUEST },
+        { type: types.SEARCH_ERROR, payload: undefined },
```

Comment:
shall we also fix the ``undefined``?

Reply:
well... we have to fix error handling for this probably. It requires changing initialState of error and dispatching search error (`searchError(error.response && error.response.data)`) and perhaps more. Let's fix that all together for all reducers/actions